vstruct/defs/macho/loader.py

```diff
@@ -118,108 +150,104 @@ def __init__(self):
         self.reserved2    = v_uint32() # reserved (for count or sizeof)
         self.reserved3    = v_uint32() # reserved
 
-class fvmlib_command(vstruct.VStruct):
+class fvmlib_command(load_command):
 
     def __init__(self):
-        vstruct.VStruct.__init__(self)
-        self.cmd     = v_uint32() # LC_IDFVMLIB or LC_LOADFVMLIB
-        self.cmdsize = v_uint32() # includes pathname string
+        # LC_IDFVMLIB or LC_LOADFVMLIB
+        # includes pathname string
+        load_command.__init__(self)
         self.name          = lc_str() # library's target pathname
         self.minor_version = v_uint32() # library's minor version number
         self.header_addr   = v_uint32() # library's header address
 
-class dylib_command(vstruct.VStruct):
+    def pcb_name(self):
+        pass
+
+class dylib_command(load_command):
 
     def __init__(self):
-        vstruct.VStruct.__init__(self)
-        self.cmd     = v_uint32() # LC_ID_DYLIB, LC_LOAD_{,WEAK_}DYLIB, LC_REEXPORT_DYLIB
-        self.cmdsize = v_uint32() # includes pathname string
+        # LC_ID_DYLIB, LC_LOAD_{,WEAK_}DYLIB, LC_REEXPORT_DYLIB
+        # includes pathname string
+        load_command.__init__(self)
         self.name                  = lc_str() # library's path name
         self.timestamp             = v_uint32() # library's build time stamp
         self.current_version       = v_uint32() # library's current version number
         self.compatibility_version = v_uint32() # library's compatibility vers number
-        self.namedata              = v_bytes(size=0)
+        self.namedata              = ''
```

Comment:
ok, having struggles with this one now that i'm thinking through it... what if we happen to want to vsEmit this?  why the switch to a raw string?

Reply:
an example of what i'm talking about:
```
In [2]: from vstruct.defs.macho import *                                                                                                                                                                                                

In [3]: m = mach_o() 

In [6]: m.vsParse(open('osx/amd64/MachO-OSX-x64-ls','rb').read())                                                                                                                                                                       

In [14]: print m.load_commands.cmd13                                                                                                                                                                                                    
-------> print(m.load_commands.cmd13)
dylib_command

In [15]: print m.load_commands.cmd13.vsEmit()                                                                                                                                                                                           
-------> print(m.load_commands.cmd13.vsEmit())
b'\x0c\x00\x00\x008\x00\x00\x00\x18\x00\x00\x00\x02\x00\x00\x00\x00\x01\x9f\x00\x00\x00\x01\x00'

In [16]: print m.load_commands.cmd13.tree()                                                                                                                                                                                             
-------> print(m.load_commands.cmd13.tree())
00000000 (24) dylib_command: dylib_command
00000000 (04)   cmd: 0x0000000c (12)
00000004 (04)   cmdsize: 0x00000038 (56)
00000008 (04)   name: 0x00000018 (24)
0000000c (04)   timestamp: 0x00000002 (2)
00000010 (04)   current_version: 0x009f0100 (10420480)
00000014 (04)   compatibility_version: 0x00010000 (65536)


In [17]: print m.load_commands.cmd13.namedata                                                                                                                                                                                           
-------> print(m.load_commands.cmd13.namedata)
/usr/lib/libSystem.B.dylib
```

Reply:
<facepalm>  now i see it.  duh.  that string comes from elsewhere, and isn't really part of the parsing.  nm!  brain coming on line in 5, 4, 3, 2, ...